astropy/visualization/wcsaxes/tests/test_misc.py

```diff
@@ -499,6 +500,37 @@ def test_simplify_labels_usetex(ignore_matplotlibrc, tmp_path):
     fig.savefig(tmp_path / "plot.png")
 
 
+@pytest.mark.parametrize("usetex", [True, False])
+@pytest.mark.parametrize("unicode_minus", [True, False])
+def test_simplify_labels_minus_sign(ignore_matplotlibrc, usetex, unicode_minus):
+    # Ensure minus signs aren't removed from the front of labels across a grid of configuration possibilities
+    if usetex and TEX_UNAVAILABLE:
+        pytest.skip("TeX is unavailable")
+    with mpl.rc_context(
+        rc={"text.usetex": usetex, "axes.unicode_minus": unicode_minus}
+    ):
+        ticklabels = TickLabels(None)
+        expected_labels = []
+        for i in [1, 2]:
+            if usetex:
+                label = f"$-{i}$"
+            elif unicode_minus:
+                label = f"\N{MINUS SIGN}{i}"
+            else:
+                label = f"-{i}"
```

Comment:
I would prefer we avoid `if`s in the body of the test when we can use parametrization instead. What I'm suggesting here is to use a single `@pytest.mark.parametrize` decorator with three parameters: `usetex`, `unicode_minus`, and `expected_labels`. It be less dynamic but would isolate the test logic from the input/expectation definition.